### PR TITLE
[frontend] Manage screens when a payload is deleted 

### PR DESCRIPTION
**Pull request by savacano28**
### Proposed changes

* Manage screens when a payload is deleted (disable buttons, set the correct tag as deleted) in atomic testing, scenarios, simulations.
* Validate if an inject has a contract.

### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/1141
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/injectors/caldera/CalderaExecutor.java [hunk 1]
```diff
@@ -84,81 +84,84 @@ public ExecutionProcess process(@NotNull final Execution execution, @NotNull fin
...
-                execution.addTrace(traceError("Caldera failed to execute the ability on asset " + asset.getName() + " (" + e.getMessage() + ")"));
-                log.severe(Arrays.toString(e.getStackTrace()));
-            }
-        });
-
-        List<AssetGroup> assetGroups = injection.getAssetGroups();
-        assetGroups.forEach((assetGroup -> computeExpectationsForAssetGroup(expectations, content, assetGroup, new ArrayList<>())));
+            });
+        }else{
+            execution.addTrace(traceError("Inject has not a contract"));
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/injectors/caldera/CalderaExecutor.java:156]: Is it better English?

Inject does not have a contract

File: openbas-api/src/main/java/io/openbas/injectors/caldera/CalderaExecutor.java [hunk 2]
```diff
@@ -84,81 +84,84 @@ public ExecutionProcess process(@NotNull final Execution execution, @NotNull fin
             execution.addTrace(traceError("Found 0 asset to execute the ability on (likely this inject does not have any target or the targeted asset is inactive and has been purged)"));
         }
         String contract;
-        if (inject.getInjectorContract().getPayload() != null) {
-            // This is a payload, need to create the ability on the fly
-            List<Ability> abilities = calderaService.abilities().stream().filter(ability -> ability.getName().equals(inject.getInjectorContract().getPayload().getId())).toList();
-            if (!abilities.isEmpty()) {
-                calderaService.deleteAbility(abilities.getFirst());
+        if(inject.hasInjectorContract()) {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/injectors/caldera/CalderaExecutor.java:87]: Wouldn't it be simpler to manage if getInjectorContract returned an optional ?

File: openbas-front/src/admin/components/common/injects/Injects.js [hunk 3]
```diff
@@ -487,11 +490,14 @@ const Injects = (props) => {
...
-                          label={injectorContractName}
-                        />
+                        {injectContract ? (
+                          <InjectorContract
+                            variant="list"
+                            config={injectContract?.config}
+                            label={injectorContractName}
+                          />
+                        ) : <InjectorContract variant="list" label={t('Deleted')} deleted={true} />
+                        }
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/common/injects/Injects.js:500]: Config is not used in the InjectorContract component
You can avoid {true] in react, it's the default value.

```suggestion
                        {injectContract ? (
                          <InjectorContract
                            variant="list"
                            label={injectorContractName}
                          />
                        ) : <InjectorContract variant="list" label={t('Deleted')} deleted />
                        }
```

File: openbas-front/src/admin/components/atomic_testings/InjectList.tsx [hunk 4]
```diff
@@ -190,17 +181,13 @@ const InjectList: FunctionComponent<Props> = ({
...
           return (
-
             <ListItem
               key={injectDto.inject_id}
               classes={{ root: classes.item }}
               divider
               secondaryAction={
                 <AtomicTestingPopover
                   atomic={injectDto}
-                  entries={[{ label: 'Duplicate', action: () => handleOpenDuplicate(injectDto.inject_id) }]}
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/atomic_testings/InjectList.tsx:201]: Why this changes.
Now we have the possibility to update & delete in the list.
This is not the behavior we want:
- adding delete should be handle in this issue https://github.com/OpenBAS-Platform/openbas/issues/1205
- update is not possible in the list direclty

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-front/src/admin/components/atomic_testings/InjectList.tsx (hunk 4 above)
Comment (savacano28) [reply openbas-front/src/admin/components/atomic_testings/InjectList.tsx:201]: I'll make the changes! ty

**Review by RomuDeuxfois — APPROVED** [review]
